src/main/java/com/ericsson/eiffel/remrem/semantics/clone/PrepareLocalEiffelSchemas.java

```diff
@@ -69,26 +86,62 @@ private static void cloneEiffelRepo(String repoURL, String branch, File localEif
 
                 // To get the latest changes from remote repository.
                 localGitRepo.pull().call();
+            }
 
-                //checkout to input branch after changes pulled into local
-                localGitRepo.checkout().setName(branch).call();
+            //checkout to input branch after changes pulled into local
+            localGitRepo.checkout().setName(branch).call();
 
-            } catch (Exception e) {
-                e.printStackTrace();
-            }
+        } catch (Exception e) {
+            LOGGER.info("please check proxy settings if proxy enabled update proxy.properties file to clone behind proxy");
+            e.printStackTrace();
         }
     }
 
+    private void setProxy() {
+        ProxySelector.setDefault(new ProxySelector() {
+            @Override
+            public List<Proxy> select(URI uri) {
+                return Arrays.asList(getProxy());
+            }
+
+            @Override
+            public void connectFailed(URI uri, SocketAddress sa, IOException ioe) {
+                if (uri == null || sa == null || ioe == null) {
+                    throw new IllegalArgumentException("Arguments can not be null.");
+                }
+            }
+
+        });
+
+    }
+
+    public Proxy getProxy() {
```

Comment:
change it to private,
move the resource bundle handling to class level. and change this code accordingly.

Calling the getProxy() twice see if we can reduce the calling of this method